xclim/sdba/_adjustment.py

```diff
@@ -615,3 +615,278 @@ def extremes_adjust(
+        ref,
+        kwargs=dict(
+            bin_width=bin_width,
+            bin_origin=bin_origin,
+            numItermax=numItermax,
+        ),
+        input_core_dims=[["time_hist", "multivar"], ["time_ref", "multivar"]],
+        output_core_dims=[["time_hist", "multivar"]],
+        keep_attrs=True,
+        vectorize=True,
+    ).rename("scen", time_hist="time")
+
+    return scen.to_dataset()
+
+
+def _dotc_adjust(
+    X1: np.ndarray,
+    Y0: np.ndarray,
+    X0: np.ndarray,
+    bin_width: np.ndarray | None = None,
+    bin_origin: np.ndarray | None = None,
+    numItermax: int | None = 100_000_000,
+    cov_factor: str | None = "std",
+):
+    """Dynamical Optimal Transport Correction of the bias of X with respect to Y.
+
+    Parameters
+    ----------
+    X1 : np.ndarray
+        Simulation data to adjust.
+    Y0 : np.ndarray
+        Bias correction reference.
+    X0 : np.ndarray
+        Historical simulation data.
+    bin_width : np.ndarray | None
+        Bin widths for all dimensions.
+    bin_origin : np.ndarray | None
+        Bin origins for all dimensions.
+    numItermax : int | None
+        Number of iterations of the earth mover distance algorithm.
+    cov_factor : str | None = "std"
+        Rescaling factor.
+
+    Returns
+    -------
+    np.ndarray
+        Adjusted data
+    """
+    # Remove NaNs
+    Y0 = Y0[~np.isnan(Y0).any(axis=1), :]
+    X0 = X0[~np.isnan(X0).any(axis=1), :]
+    X1 = X1[~np.isnan(X1).any(axis=1), :]
+
+    # Initialize parameters
+    bin_width = u.bin_width_estimator([Y0, X0, X1]) if bin_width is None else bin_width
+    if cov_factor == "cholesky":
+        fact0 = u.eps_cholesky(np.cov(Y0, rowvar=False))
+        fact1 = u.eps_cholesky(np.cov(X0, rowvar=False))
+        cov_factor = np.dot(fact0, np.linalg.inv(fact1))
+    elif cov_factor == "std":
+        fact0 = np.std(Y0, axis=0)
+        fact1 = np.std(X0, axis=0)
+        cov_factor = np.diag(fact0 / fact1)
+    else:
+        cov_factor = np.identity(Y0.shape[1])
+
+    # Map ref to hist
+    yX0 = _otc_adjust(
+        Y0, X0, bin_width=bin_width, bin_origin=bin_origin, numItermax=numItermax
+    )
+
+    # Map hist to sim
+    yX1 = _otc_adjust(
+        X0,
+        X1,
+        src=yX0,
+        bin_width=bin_width,
+        bin_origin=bin_origin,
+        numItermax=numItermax,
+    )
+
+    # Temporal evolution
+    motion = yX1 - yX0
+    # Apply a variance dependent rescaling factor
+    motion = np.apply_along_axis(lambda x: np.dot(cov_factor, x), 1, motion)
+
+    # Apply the evolution to ref
+    Y1 = Y0 + motion
+
+    # Map sim to the evolution of ref
+    Z1 = _otc_adjust(
+        X1, Y1, bin_width=bin_width, bin_origin=bin_origin, numItermax=numItermax
+    )
+
+    return Z1
+
+
+@map_groups(scen=[Grouper.DIM])
+def dotc_adjust(
+    ds: xr.Dataset,
+    dim: str,
+    bin_width: np.ndarray | None = None,
+    bin_origin: np.ndarray | None = None,
+    numItermax: int | None = 100_000_000,
```

Comment:
We usually use snake_case convention
```suggestion
    num_iter_max: int | None = 100_000_000,
```
Also @aulemahal do we have other instances where we can give the power to users to control how long we should try to converge an algorithm? I don't think so.  I guess this could be compared to the number of rotations in MBCn, we could in principle set a target error value that we want to reach for convergence with a maximal number of rotations in MBCn. I think it's OK to give this control to users, thoughts?